### PR TITLE
chore: migrate HiddenControl component from jsx to tsx

### PR DESCRIPTION
**Pull request by Damans227**
### SUMMARY

PR for migrating `HiddenControl` functional react component from JSX to TSX

### ADDITIONAL INFORMATION

- [x] Has associated issue: [TRACKER] Migrate JavaScript files to TypeScript #10004
- [x] Enhancement (new features, refinement)
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by Damans227** [comment]
@etr2460 Ptal, thanks!

File: superset-frontend/src/explore/components/controls/HiddenControl.tsx [hunk 1]
```diff
@@ -17,29 +17,14 @@
...
-    PropTypes.func,
-  ]),
-};
-
-const defaultProps = {
-  onChange: () => {},
-};
+interface HiddenControlsProps {
+  onChange: () => void;
+  value: string | number | {} | boolean | [] | (() => void);
```
Comment (etr2460) [inline superset-frontend/src/explore/components/controls/HiddenControl.tsx:24]: I _think_ this should probably be

```suggestion
  value: string | number | Record<string, any> | boolean | any[] | Function;
```

Comment (Damans227) [reply superset-frontend/src/explore/components/controls/HiddenControl.tsx]: Thanks for the feedback. I have committed the suggested changes, and will keep them in mind for future.

**Comment by etr2460** [comment]
@Damans227 looks like there's a couple lint issues here. I'd recommend running `npm run type` locally to see the typescript errors, and also updating the pr title according to the PR Lint error here

**Comment by Damans227** [comment]
> @Damans227 looks like there's a couple lint issues here. I'd recommend running `npm run type` locally to see the typescript errors, and also updating the pr title according to the PR Lint error here

@etr2460 Looks like there is a type mismatch: 

```
src/explore/components/controls/HiddenControl.tsx:29:31 - error TS2322: Type 'string | number | boolean | Function | any[] | Record<string, any>' is not assignable to type 'string | number | readonly string[] | undefined'.
  Type 'false' is not assignable to type 'string | number | readonly string[] | undefined'.

29   return <Input type="hidden" value={props.value} />;
                                 ~~~~~

  node_modules/@types/react/index.d.ts:2224:9
    2224         value?: string | ReadonlyArray<string> | number | undefined;
                 ~~~~~
    The expected type comes from property 'value' which is declared here on type 'IntrinsicAttributes & Pick<InputProps, "dir" | "form" | "slot" | "style" | "title" | "pattern" | "id" | "children" | "prefixCls" | "className" | ... 281 more ... | "bordered"> & ... 4 more ... & { ...; }'

```
If I change the type of value in the interface to `value: 'string | number | readonly string[] | undefined';` then there are no errors.

**Comment by etr2460** [comment]
that sounds good to me

File: superset-frontend/src/explore/components/controls/HiddenControl.tsx [hunk 2]
```diff
@@ -17,29 +17,14 @@
...
-    PropTypes.func,
-  ]),
-};
-
-const defaultProps = {
-  onChange: () => {},
-};
+interface HiddenControlsProps {
+  onChange: () => void;
+  value: 'string | number | readonly string[] | undefined';
```
Comment (etr2460) [inline superset-frontend/src/explore/components/controls/HiddenControl.tsx:24]: this probably shouldn't be a string

```suggestion
  value: string | number | readonly string[] | undefined;
```

Comment (Damans227) [reply superset-frontend/src/explore/components/controls/HiddenControl.tsx]: That's right. Thanks for the suggestion.

**Comment by Damans227** [comment]
> @etr2460 Ephemeral environment spinning up at http://54.191.46.36:8080. Credentials are `admin`/`admin`. Please allow several minutes for bootstrapping and startup.

@etr2460  Looks ok to me. 

Screen Shot 2021-11-02 at 1 27 43 PM

**Review by etr2460 — APPROVED** [review]

**Pull request merged**